subprojects/core-api/src/main/java/org/gradle/api/artifacts/transform/TransformAction.java

```diff
@@ -16,20 +16,41 @@
 
 package org.gradle.api.artifacts.transform;
 
+import org.gradle.api.Action;
 import org.gradle.api.Incubating;
 
 import javax.inject.Inject;
 
 /**
  * Interface for artifact transform actions.
  *
- * <p>Implementations must provide a public constructor.</p>
+ * <p>
+ *     A transform action implementation is an abstract class implementing the {@link #transform(TransformOutputs)} method.
+ *     The abstract methods will be created by Gradle when possible.
```

Comment:
```suggestion
 *     Gradle will provide implementations for any abstract method when possible.
```

Is this for any "method" or only for getters?

Also, "when possible" is probably not a good phrasing here, as "when not possible" is an actual error.

Reply:
I removed the phrase, given that we say later on:
```
Implementations can receive parameters by using annotated abstract getter methods.
```